libraries/engage/page/components/Widgets/Widget.jsx

```diff
@@ -132,6 +132,9 @@ const Widget = ({
             {...(isCustomLegacyWidget ? {
               settings: definition.widgetConfig,
             } : {})}
+            {...(isPreview ? {
```

Comment:
i think it's not a good idea to pass a conditional click listener here. the only api that is provided to the widgets should be the `useWidgetHook`.
please check the comments for the `Button` before you answer.